src/controller/python/chip/yaml/runner.py

```diff
@@ -0,0 +1,387 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from chip import ChipDeviceCtrl
+from enum import Enum
+import stringcase
+import chip.interaction_model
+import asyncio as asyncio
+import logging
+from chip.yaml.errors import ParsingError, UnexpectedParsingError
+from chip.clusters.Attribute import AttributeStatus, ValueDecodeFailure
+from .data_model_lookup import *
+import chip.yaml.format_converter as Converter
+
+logger = logging.getLogger('YamlParser')
+
+
+class _ActionStatus(Enum):
+    SUCCESS = 'success',
+    ERROR = 'error'
+
+
+@dataclass
+class _ActionResult:
+    status: _ActionStatus
+    response: object
+
+
+@dataclass
+class _ExecutionContext:
+    ''' Objects that is commonly passed around this file that are vital to test execution.'''
+    # Data model lookup to get python attribute, cluster, command object.
+    data_model_lookup: DataModelLookup = None
+
+
+class BaseAction(ABC):
+    '''Interface for a single YAML action that is to be executed.'''
+
+    def __init__(self, label):
+        self._label = label
+
+    @property
+    def label(self):
+        return self._label
+
+    @abstractmethod
+    def run_action(self, dev_ctrl: ChipDeviceCtrl) -> _ActionResult:
+        pass
+
+
+class InvokeAction(BaseAction):
+    '''Single invoke action to be executed.'''
+
+    def __init__(self, test_step, cluster: str, context: _ExecutionContext):
+        '''Converts 'test_step' to invoke command action that can execute with ChipDeviceCtrl.
+
+        Args:
+          'test_step': Step containing information required to run invoke command action.
+          'cluster': Name of cluster which to invoke action is targeting.
+          'context': Contains test-wide common objects such as DataModelLookup instance.
+        Raises:
+          ParsingError: Raised if there is a benign error, and there is currently no
+            action to perform for this write attribute.
+          UnexpectedParsingError: Raised if there is an unexpected parsing error.
+        '''
+        super().__init__(test_step.label)
+        self._command_name = stringcase.pascalcase(test_step.command)
+        self._cluster = cluster
+        self._request_object = None
+        self._expected_response_object = None
+        self._endpoint = test_step.endpoint
+        self._node_id = test_step.node_id
+
+        command = context.data_model_lookup.get_command(self._cluster, self._command_name)
+
+        if command is None:
+            raise ParsingError(
+                f'Failed to find cluster:{self._cluster} Command:{self._command_name}')
+
+        command_object = command()
+        if (test_step.arguments):
+            args = test_step.arguments['values']
```

Comment:
If `test_step.arguments` is not None this will always have 'values' even just a single value will be inside `values` this is something that common parser guarantees. I will add a comment here